doozer/doozerlib/build_info.py

```diff
@@ -474,8 +604,63 @@ async def find_non_latest_rpms(self, arch_rpms_to_check: Optional[Dict[str, List
         arches = meta.get_arches()
         tasks: List[Awaitable[List[Tuple[str, str, str]]]] = []
         for arch in arches:
-            iar = self.get_image_archive_inspector(arch)
+            iar = self.get_image_inspector(arch)
             assert iar is not None
             tasks.append(iar.find_non_latest_rpms(arch_rpms_to_check[arch] if arch_rpms_to_check else None))
         result = dict(zip(arches, await asyncio.gather(*tasks)))
         return result
+
+
+class KonfluxBuildRecordInspector(BuildRecordInspector):
+    def __init__(self, runtime, build_record: KonfluxBuildRecord):
+        super().__init__(runtime)
+        self._build_record = build_record
+        self._nvr = build_record.nvr
+        self.image_pullspec = build_record.image_pullspec
+        self._inspectors = []
+
+    def get_build_id(self):
+        return self._build_record.build_id
+
+    def get_build_obj(self):
+        return self._build_record
+
+    def get_build_pullspec(self):
+        return self._build_record.image_pullspec
+
+    def is_under_embargo(self):
+        return self._build_record.embargoed
+
+    def get_version(self):
+        return self._build_record.version
+
+    def get_release(self):
+        return self._build_record.release
+
+    def get_all_installed_package_build_dicts(self):
+        raise NotImplementedError
+
+    def get_rpms_in_pkg_build(self, build_id: int):
+        raise NotImplementedError
+
+    def get_image_meta(self):
+        return self.runtime.image_map[self.get_build_obj().name]
+
+    def get_image_inspectors(self):
+        if not self._inspectors:
+            info = util.oc_image_info_show_multiarch__caching(
+                pullspec=self.get_build_pullspec(),
+                registry_username=os.environ['KONFLUX_ART_IMAGES_USERNAME'],
+                registry_password=os.environ['KONFLUX_ART_IMAGES_PASSWORD']
+            )
+            if isinstance(info, dict):
+                # The pullspec points to a single arch image
+                self._inspectors.append(KonfluxImageInspector(self.runtime, info, self))
+            else:
+                # The pullspec points to a multi arch manifest list
+                self._inspectors.extend(
+                    [KonfluxImageInspector(self.runtime, item, self) for item in info])
+        return self._inspectors
+
+    def get_build_webpage_url(self):
+        return 'TODO'  # TODO art-dash should display build info for a given build ID
```

Comment:
i think for now this could simply be the url of the plr run that we store in db?

Reply:
Done with a3fe10d7dfc29c989de349e80703e741c7459ded